cmd/collector/app/server/grpc.go

```diff
@@ -42,31 +42,40 @@ type GRPCServerParams struct {
 // StartGRPCServer based on the given parameters
 func StartGRPCServer(params *GRPCServerParams) (*grpc.Server, error) {
 	var server *grpc.Server
-	var grpcOpts []grpc.ServerOption
+	var grpcOpts []configgrpc.ToServerOption
 
-	if params.MaxReceiveMessageLength > 0 {
-		grpcOpts = append(grpcOpts, grpc.MaxRecvMsgSize(params.MaxReceiveMessageLength))
+	if params.MaxRecvMsgSizeMiB > 0 {
+		grpcOpts = append(grpcOpts, configgrpc.WithGrpcServerOption(grpc.MaxRecvMsgSize(params.MaxRecvMsgSizeMiB)))
+	}
+	if params.Keepalive != nil {
+		grpcOpts = append(grpcOpts, configgrpc.WithGrpcServerOption(grpc.KeepaliveParams(keepalive.ServerParameters{
+			MaxConnectionAge:      params.Keepalive.ServerParameters.MaxConnectionAge,
+			MaxConnectionAgeGrace: params.Keepalive.ServerParameters.MaxConnectionAgeGrace,
+		})))
```

Comment:
these parameters are part of the config that you are using to create the server (via `params.ToServer()`). So I am sure they are already being taken care of by `ToServer()`, there is not need to manually transform them into options.